unit/memory-analyzer/gdb_api.cpp

```diff
@@ -104,94 +104,111 @@ void gdb_api_internals_test()
 TEST_CASE("gdb api internals test", "[core][memory-analyzer]")
 {
 #ifdef RUN_GDB_API_TESTS
-  gdb_api_internals_test();
+  const bool has_gdb = system("gdb --version") == 0;
+  SECTION("check gdb is on the PATH")
+  {
+    REQUIRE(has_gdb);
+  }
+  if(has_gdb)
+  {
+    gdb_api_internals_test();
+  }
 #endif
 }
 
 TEST_CASE("gdb api test", "[core][memory-analyzer]")
 {
 #ifdef RUN_GDB_API_TESTS
-  compile_test_file();
-
+  const bool has_gdb = system("gdb --version") == 0;
```

Comment:
As above - or maybe even factor out this check (and the `SECTION`) into a separate procedure.

Reply:
Done.